File: sanghoon/programmers-2/N개의 최소공배수.py
def solution(arr):
    answer = max(arr)
    cnt = 0
    while cnt != len(arr):
        cnt = 0
        for i in arr :
            if answer % i == 0 :
                cnt += 1
            else :
                answer += 1
                break
    return answer

arr = [1,2,3]
solution(arr)

File: kyulee/programmers-2/N개의_최소공배수.java
public class N개의_최소공배수 {
    class Solution {
        public int solution(int[] arr) {
            int answer = arr[0];

            for (int num : arr) {
                answer = answer * num / gcd(answer, num);
            }
            return answer;
        }

        public int gcd(int a, int b) {
            while (b != 0) {
                int temp = b;
                b = a % b;
                a = temp;
            }
            return a;
        }
    }
}
